x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/local/ReplaceMissingFieldWithNull.java

```diff
@@ -66,37 +79,29 @@ private LogicalPlan missingToNull(LogicalPlan plan, SearchStats stats, Attribute
         else if (plan instanceof Project project) {
             var projections = project.projections();
             List<NamedExpression> newProjections = new ArrayList<>(projections.size());
-            Map<DataType, Alias> nullLiteral = Maps.newLinkedHashMapWithExpectedSize(DataType.types().size());
             AttributeSet joinAttributes = joinAttributes(project);
+            ArrayList<Alias> replacedFieldAttributesInProject = new ArrayList<>();
 
             for (NamedExpression projection : projections) {
-                // Do not use the attribute name, this can deviate from the field name for union types.
                 if (projection instanceof FieldAttribute f
+                    // Do not use the attribute name, this can deviate from the field name for union types
                     && stats.exists(f.fieldName()) == false
                     && joinAttributes.contains(f) == false
                     && f.field() instanceof PotentiallyUnmappedKeywordEsField == false) {
                     // TODO: Should do a searchStats lookup for join attributes instead of just ignoring them here
                     // See TransportSearchShardsAction
-                    DataType dt = f.dataType();
-                    Alias nullAlias = nullLiteral.get(f.dataType());
-                    // save the first field as null (per datatype)
-                    if (nullAlias == null) {
-                        Alias alias = new Alias(f.source(), f.name(), Literal.of(f, null), f.id());
-                        nullLiteral.put(dt, alias);
-                        projection = alias.toAttribute();
-                    }
-                    // otherwise point to it
-                    else {
```

Comment:
This was wrong to begin with. There is no guarantee that an alias for a keyword-null created once can even be accessible by upstream commands, or other branches of binary plans.